docs/projects/project-setup.md

```diff
@@ -0,0 +1,217 @@
+# MLRun project bootstrapping with `project_setup.py`
+
+## Overview
+
+The `project_setup.py` script in MLRun automates project initialization and configuration, facilitating seamless setup of MLRun projects by registering functions, workflows, Git sources, Docker images, and more. It ensures consistency by registering and updating all functions and workflows within the project.
+
+Upon loading an MLRun project via via {py:meth}`~mlrun.projects.get_or_create_project` or {py:meth}`~mlrun.projects.load_project`, the system automatically invokes the `project_setup.py` script.
```

Comment:
Upon loading an MLRun project via {py:meth}`~mlrun.projects.get_or_create_project` or {py:meth}`~mlrun.projects.load_project`, the system automatically invokes the `project_setup.py` script.